src/ctia/domain/id.clj

```diff
@@ -70,13 +73,14 @@
 
 (defn short-id->id
   [type short-id {:keys [hostname path-prefix port protocol]}]
-  (map->CtiaId
-   {:hostname hostname
-    :short-id short-id
-    :path-prefix (not-empty path-prefix)
-    :port port
-    :protocol protocol
-    :type (name type)}))
+  (if (valid-short-id? short-id)
```

Comment:
I believe the first reason is because I also code in Haskell I believe. And in Haskell you are forced to provide both values which are also statically typed to have the same type.
If you want to make a one way it is more explicit. So in my habits, when I see an `if` I tend to search the else value while when I see `when` it is easier for me to read.

Also, if you treat your `if` as a `when`, adding another effectful line (like a println) won't change the meaning. But this is mainly a matter of taste.